File: Simulacao/MM1_Python/Discreta.py
from FGVA import fgva

class discreta(fgva):
    def __init__(self, a, probA, b, probB, c, probC):
        fgva.__init__(self)
        self.a = a
        self.probA = probA
        self.b = b
        self.probB = probB
        self.c = c
        self.probC = probC
    def gerarNumero(self):
        randomNum = fgva.gerarNumero(self)
        if randomNum <= probA:
            return a
        elif randomNum <= probB:
            return b
        else:
            return c

File: Simulacao/MM_2/distribuicao/discreta.js
class Discreta extends FVGA {

    constructor(A, probA, B, probB, C, probC) {
        super();
        this.A = A;
        this.probA = probA;
        this.B = B;
        this.probB = probB;
        this.C = C;
        this.probC = probC;
    }

    gerarNumero() {
        const randomNum = super.gerarNumero();

        if (randomNum <= this.probA) {
            return this.A;
        }
        else if (randomNum <= this.probB) {
            return this.B;
        }
        else {
            return this.C;
        }
    }
}